### PR TITLE
fix: Use VirtualFiles to save dot file, improved logging

### LINKED ISSUE #47: [Error report] java.lang.Throwable: Cannot create .copilot-plugin
Plugin version: **1.0.9**IDE version: **IntelliJ IDEA 2023.2.4**OS: **Windows 11**Stacktrace:```java.lang.Throwable: Cannot create .copilot-pluginat com.intellij.openapi.diagnostic.Logger.error(Logger.java:370)at com.vaadin.plugin.copilot.CopilotPluginUtil$Companion.savePortInDotFile(CopilotPluginUtil.kt:182)at com.vaadin.plugin.copilot.CopilotPluginUtil$Companion.startServer(CopilotPluginUtil.kt:109)at com.vaadin.plugin.copilot.activity.CopilotPostStartupProjectActivity.execute(CopilotPostStartupProjectActivity.kt:18)at com.intellij.ide.startup.impl.StartupManagerImplKt$launchActivity$1$1.invokeSuspend(StartupManagerImpl.kt:534)at com.intellij.ide.startup.impl.StartupManagerImplKt$launchActivity$1$1.invoke(StartupManagerImpl.kt)at com.intellij.ide.startup.impl.StartupManagerImplKt$launchActivity$1$1.invoke(StartupManagerImpl.kt)at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:167)at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)at com.intellij.ide.startup.impl.StartupManagerImplKt$launchActivity$1.invokeSuspend(StartupManagerImpl.kt:532)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)```

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

Use VirtualFile to save `.copilot-plugin.properties` file.

Based on [contribution](https://github.com/vaadin/intellij-plugin/issues/47#issuecomment-2386479560) by @sveinnetnordic

Fixes #47

### CODE REVIEW

**Comment by sveinnetnordic** [comment]
> Artifact build on last commit: [distributions.zip](https://github.com/vaadin/intellij-plugin/actions/runs/11141571688/artifacts/2005130005). For MacOS users: there is a zip inside this zip and Finder unzips them both at once. Use `unzip distributions.zip` from Terminal or [check solution for Archive Manager](https://apple.stackexchange.com/questions/443607/is-there-a-way-to-prevent-macoss-archive-utility-from-unarchiving-inner-zip-fil).

This works for me :-)

File: src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt [hunk 1]
```diff
@@ -95,53 +91,61 @@ class CopilotPluginUtil {
...
-            WriteCommandAction.runWriteCommandAction(project) {
-                val ideaDir = getIdeaDir(project).path
-                VfsUtil.createDirectoryIfMissing(ideaDir)?.let { LOG.info("$ideaDir created") }
+        private fun getDotFileDirectory(project: Project): VirtualFile? {
+            val projectDir = project.guessProjectDir()
+            if (projectDir == null) {
+                LOG.error("Cannot guess project directory")
+                return null
             }
+            LOG.info("Project directory: $projectDir")
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt:147]: Is this info or debug info?

File: src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt [hunk 2]
```diff
@@ -95,53 +91,61 @@ class CopilotPluginUtil {
...
+                                LOG.error("Failed to create $DOTFILE: ${e.message}")
+                                return@runWriteAction
+                            }
+
+                        try {
+                            VfsUtil.saveText(newFile, stringWriter.toString())
+                        } catch (e: IOException) {
+                            LOG.error("Failed to write to $DOTFILE: ${e.message}")
+                        }
+                        LOG.info("$newFile created")
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt:117]: Is this info or debug info?

**Review by Artur- — APPROVED** [review]
Just a few questions/nits about logging

File: src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt (hunk 2 above)
Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt:117]: According to https://plugins.jetbrains.com/docs/intellij/ide-infrastructure.html#logging `INFO` and above goes to `idea.log` by default. As I'm looking into that log those entries fits perfectly there. It's information about work of plugin so I would keep it as `INFO`.

**Pull request merged**